tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/internal/buffer/ByteSequencer.java

```diff
@@ -0,0 +1,53 @@
+package org.tensorflow.internal.buffer;
+
+import java.util.Iterator;
+import java.util.function.Function;
+import org.tensorflow.ndarray.NdArray;
+import org.tensorflow.ndarray.NdArraySequence;
+
+/**
+ * Produces sequence of bytes to be stored in a {@link ByteSequenceTensorBuffer}.
+ *
+ * @param <T> source of bytes (byte arrays or strings)
+ */
+public class ByteSequencer<T> implements Iterable<byte[]> {
```

Comment:
Maybe `ByteArrayProducer`? `ByteSequencer` sounds like it can change the sequence but my reading of the iterator is that the iteration order is fixed.

Reply:
Yeah, I wasn't sure about this one neither... I was hesitating between this and `ByteSequenceProducer` so sounds like the second choice would be a good trade off? (I prefer `ByteSequence` to `ByteArray` to preserve the symmetry with `ByteSequenceTensorBuffer`, or I rename the latter to `ByteArrayTensorBuffer`?).

Reply:
I think `ByteSequenceProducer` works.